cpp/src/arrow/csv/writer_test.cc

```diff
@@ -171,6 +173,13 @@ std::vector<WriterTestParams> GenerateTestCases() {
         "style is \"None\". See RFC4180. Invalid value: ",
         value);
   };
+
+  auto expected_status_illegal_delimiter = [](const char value) {
+    return Status::Invalid(
+        "WriteOptions: delimiter cannot be \\r or \\n or \" or EOL. Invalid value: ",
+        value);
+  };
```

Comment:
Move it under line 205 in delimiter test section.

Reply:
Made the change.